util/rate_limiter.cc

```diff
@@ -111,7 +111,8 @@ void GenericRateLimiter::Request(int64_t bytes, const Env::IOPriority pri,
     std::chrono::microseconds now(NowMicrosMonotonic(env_));
     if (now - tuned_time_ >=
         kRefillsPerTune * std::chrono::microseconds(refill_period_us_)) {
-      Tune();
+      Status s = Tune();
+      assert(s.ok());  //**TODO: What to do on error?
```

Comment:
At the moment `GenericRateLimiter::Tune` will always return `Status::OK`. So I guess the question is whether the rate limiter API should be modified to return `void` or not?

If not, I cannot see a good solution if the status is NOK apart from logging a message and returning.

Reply:
@pdillinger Should the Request APIs into the RateLimiter be changed to return a Status?  It would be a public API change, but would not require anything on the user's part, unless they implemented their own RateLimiter.